### PR TITLE
Generate Step description in HTML

### PR DESCRIPTION
**Pull request by alecharp**
Generating the step description in HTML provide a better control on how the elements are indented and displayed.

@reviewbybees esp @jglick 

This is providing a better visualization of the step parameters and configuration. From:

2018-02-05-15-32-44_849x1088

to 

2018-02-05-15-33-04_869x1121

### CODE REVIEW

File: src/main/java/org/jenkinsci/pipeline_steps_doc_generator/ToAsciiDoc.java [hunk 1]
```diff
@@ -36,62 +38,63 @@ private static String header(int depth){
         return String.join("", Collections.nCopies(depth, "="));
     }
 
-    private static String listDepth(int depth){
-        return String.join("", Collections.nCopies(depth+1, ":"));
-    }
-
     private static String helpify(String help){
-        return "++++\n"+ Jsoup.clean(help, Whitelist.relaxed().addEnforcedAttribute("a", "rel", "nofollow"))+"\n++++\n";
+        return Jsoup.clean(help, Whitelist.relaxed().addEnforcedAttribute("a", "rel", "nofollow")) + "\n";
```
Comment (jglick) [inline src/main/java/org/jenkinsci/pipeline_steps_doc_generator/ToAsciiDoc.java]: recommend wrapping in `<div>`

**Comment by ndeloof** [comment]
@alecharp ❤️. Wish you can do the same for configuration-as-code ref documentation

**Comment by kwhetstone** [comment]
The hardest part of that AsciiDoc was trying to get the indents and the spacing right since it seemed like if there weren't enough returns everything would bunch together.  HTML is so much easier to generate and set up right.

**Comment by jglick** [comment]
Note that @alecharp was researching how to put stuff into collapsible sections (preferably collapsed by default), like we do in the live product. But that could be filed as a follow-up; certainly this is a step forward.

**Pull request merged**